### PR TITLE
feat: propagate `pnpm-workspace.yaml` load errors

### LINKED ISSUE #402: Propagate `pnpm-workspace.yaml` load errors instead of silently swallowing them
## Background

pnpm v11 reads almost all configs from `pnpm-workspace.yaml` and reserves `.npmrc` for auth and network keys. The split lives in
[`isNpmrcReadableKey`](https://github.com/pnpm/pnpm/blob/8eb1be4988/config/reader/src/localConfig.ts) and the YAML loader is
[`workspace-manifest-reader`](https://github.com/pnpm/pnpm/blob/8eb1be4988/workspace/workspace-manifest-reader/src/index.ts).

Pacquet *already* reads `pnpm-workspace.yaml`:

- `crates/npmrc/src/workspace_yaml.rs` defines `WorkspaceSettings` mirroring the pnpm‑11 camelCase key set (`storeDir`, `nodeLinker`, hoist family, `lockfile`, `fetchRetries*`, `verifyStoreIntegrity`, …).
- `Npmrc::current` walks up from cwd to find the nearest `pnpm-workspace.yaml` and layers its overrides on top of the `.npmrc`-derived `registry`.
- `crates/npmrc/src/npmrc_auth.rs` keeps only `registry` from `.npmrc`; project-structural keys there are silently ignored.
- Existing tests already exercise this: `pnpm_workspace_yaml_registry_overrides_npmrc_registry`, `pnpm_workspace_yaml_found_by_walking_up`, `non_auth_keys_in_npmrc_are_ignored`, `fetch_retry_keys_in_npmrc_are_ignored`, plus the suite in `workspace_yaml/tests.rs`.

The remaining divergence from pnpm is on the **error path**. Upstream's `readManifestRaw` returns `undefined` only for `ENOENT` and propagates every other failure (read failure, invalid yaml). Pacquet swallowed all of them via `if let Ok(Some(...))`, so a typo in `pnpm-workspace.yaml` silently fell through to defaults — leaving the user uncertain which install settings were actually in effect.

## Tasks

- [x] Make `Npmrc::current` return `Result<Self, LoadWorkspaceYamlError>`. `find_and_load` keeps `Ok(None)` for the missing-file case (`is_file()` gate), while read and parse failures bubble up.
- [x] Wrap the new error with `miette::Diagnostic` so the CLI prints a clean message including the offending path. Box the variant payloads to keep the `Result` small (clippy `result_large_err`).
- [x] Update CLI call sites and `StoreCommand::run` to thread the new fallible signature.
- [x] Add a regression test (`invalid_workspace_yaml_propagates_error`) asserting invalid yaml surfaces as `ParseYaml { .. }` instead of being silently dropped.

## Requirements

Mirror pnpm v11.

Carefully read and comply with `AGENTS.md`, `CONTRIBUTING.md`, and `CODE_STYLE_GUIDE.md` before writing code. Also, after writing code, carefully double check whether the written code complies with these files.

Carefully read [the DI issue](https://github.com/pnpm/pacquet/issues/339) and abide by it IF applicable.

---
Written by an agent (Claude Code, claude-opus-4-7).

### PR DESCRIPTION
**Pull request by KSXGitHub**
Resolves <https://github.com/pnpm/pacquet/issues/402>.

## What this PR does

Pacquet already reads `pnpm-workspace.yaml` (`crates/npmrc/src/workspace_yaml.rs` defines `WorkspaceSettings` and `Npmrc::current` walks up from cwd to layer it on top of `.npmrc`). The remaining divergence from pnpm 11 was on the error path. Upstream's [`workspace-manifest-reader`](https://github.com/pnpm/pnpm/blob/8eb1be4988/workspace/workspace-manifest-reader/src/index.ts) silently treats `ENOENT` as "no manifest" and propagates every other failure (read errors, parse errors, `EISDIR`, permission denied, …). Pacquet swallowed all errors via `if let Ok(Some(...))`, and its `find_workspace_manifest` helper used an `is_file()` filter that also masked non-regular entries (e.g. a directory named `pnpm-workspace.yaml`).

This PR closes both gaps.

## Changes

- `crates/npmrc/src/workspace_yaml.rs`
  - Replace the bare `Debug`-only `LoadWorkspaceYamlError` with a `Display + Error + Diagnostic` enum that stores `path` + `source` directly on each struct variant, mirroring `pacquet_modules_yaml::ReadModulesError`. Only `serde_saphyr::Error` is boxed (`io::Error` fits the variant inline). The single-place `path` formatting keeps the miette `Caused by:` chain free of duplication.
  - Rewrite `find_and_load` as a single read-and-catch loop over `start_dir.ancestors()`: at each level read `pnpm-workspace.yaml`, treat only `ErrorKind::NotFound` as "no manifest here, walk up", and propagate every other error through `LoadWorkspaceYamlError`. This matches pnpm's `readManifestRaw` semantics, closes the previous TOCTOU window between the old `is_file()` check and the read, and surfaces non-regular entries (directory, EISDIR, permission denied) as hard errors instead of silently walking up.
- `crates/npmrc/src/lib.rs`
  - `Npmrc::current` is now `Result<Self, LoadWorkspaceYamlError>`. The `.npmrc` parsing stays best-effort; the yaml parsing fails loud.
  - Two regression tests added in this round: `invalid_workspace_yaml_propagates_error` (parse error) and `find_propagates_when_manifest_path_is_a_directory` (`EISDIR`-class error). Existing tests adopt the fallible signature with `.expect(...)`.
- `crates/cli/src/cli_args.rs`
  - The `npmrc` closure now returns `miette::Result<&'static mut Npmrc>` and is wrapped with `wrap_err("load configuration")`. The `state` closure threads `?`.
- `crates/cli/src/cli_args/store.rs`
  - `StoreCommand::run` accepts `impl FnOnce() -> miette::Result<&Npmrc>` so `Prune` and `Path` get the new error path; `Store` and `Add` still panic before loading config.
- `crates/npmrc/Cargo.toml`
  - Add `derive_more` and `miette` deps for the new error type.

## Test plan

- [x] `cargo fmt`
- [x] `cargo check --workspace --all-targets`
- [x] `cargo clippy --workspace --all-targets -- --deny warnings`
- [x] `typos`
- [x] `cargo test -p pacquet-npmrc` (37 tests, including the two new regression tests)
- [x] `cargo test -p pacquet-cli --lib` and `cargo test -p pacquet-cli --test store`
- [ ] `cargo test -p pacquet-cli` integration tests that need `just registry-mock launch` (not exercised in this sandbox; pre-existing on main)

---
Written by an agent (Claude Code, claude-opus-4-7).

## Summary by CodeRabbit

* **Bug Fixes**
  * Configuration loading now properly catches and reports errors when workspace manifest files are invalid or unreadable, preventing silent failures.
  * Enhanced error diagnostics to clearly indicate the specific configuration file location and nature of failure.

* **Tests**
  * Added tests to verify proper error propagation for malformed configuration files and edge cases in manifest loading.

### CODE REVIEW

File: crates/npmrc/src/lib.rs [hunk 1]
```diff
@@ -544,7 +555,27 @@ mod tests {
...
+    pub fn invalid_workspace_yaml_propagates_error() {
+        let tmp = tempdir().unwrap();
+        // `: : :` is rejected by saphyr.
+        fs::write(tmp.path().join("pnpm-workspace.yaml"), ": : :\n")
+            .expect("write to pnpm-workspace.yaml");
+        let result =
+            Npmrc::current(|| tmp.path().to_path_buf().pipe(Ok::<_, ()>), || None, Npmrc::new);
+        let err = result.expect_err("invalid yaml should error");
+        assert!(
+            matches!(err, crate::LoadWorkspaceYamlError::ParseYaml { .. }),
```
[the remark this answers is not in the document]
Comment (KSXGitHub) [reply crates/npmrc/src/lib.rs:577]: Addressed in faf85cb. Quick correction on the premise: in current Rust, `EnumVariant { .. }` is a valid pattern against tuple variants too — the original code did compile and the test passed locally and (per the green CI on 7077200) on GitHub. The stylistic point holds though, since `Variant(_)` is the conventional form for tuple variants, so I made the cosmetic change.

---
Written by an agent (Claude Code, claude-opus-4-7).


---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/npmrc/src/workspace_yaml.rs [hunk 2]
```diff
@@ -77,9 +106,17 @@ impl WorkspaceSettings {
...
+            .pipe_ref(fs::read_to_string)
+            .map_err(|error| ReadFileError { path: path.clone(), error })
+            .map_err(Box::new)
+            .map_err(LoadWorkspaceYamlError::ReadFile)?;
+        let settings: WorkspaceSettings = serde_saphyr::from_str(&text).map_err(|error| {
+            LoadWorkspaceYamlError::ParseYaml(Box::new(ParseYamlError {
+                path: path.clone(),
+                error,
+            }))
+        })?;
```
[the remark this answers is not in the document]
Comment (KSXGitHub) [reply crates/npmrc/src/workspace_yaml.rs]: Fixed in 2b59d70. `ErrorKind::NotFound` from `read_to_string` now short-circuits to `Ok(None)`, matching pnpm's `readManifestRaw` semantics where every `ENOENT` from the read path is treated as "no manifest". The `pipe-trait` chain that landed in 04d4333 is preserved in the non-`ENOENT` branch via a `match` wrapper.

---
Written by an agent (Claude Code, claude-opus-4-7).


---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/npmrc/src/workspace_yaml.rs [hunk 3]
```diff
@@ -56,11 +60,37 @@ pub struct WorkspaceSettings {
...
+}
+
+/// Payload of [`LoadWorkspaceYamlError::ParseYaml`].
+#[derive(Debug, Display, Error)]
+#[display("{}: {error}", path.display())]
+pub struct ParseYamlError {
+    pub path: PathBuf,
+    #[error(source)]
+    pub error: serde_saphyr::Error,
 }
```
[the remark this answers is not in the document]
Comment (KSXGitHub) [reply crates/npmrc/src/workspace_yaml.rs]: Fixed in 62eac2a. Followed option (b) since `pacquet_modules_yaml::ReadModulesError` is the established precedent: dropped the `ReadFileError` / `ParseYamlError` payload structs and put `path` + `source` directly on the struct variants. Only `serde_saphyr::Error` is large enough to need boxing; `io::Error` fits inline. The miette output is now `Failed to … at <path>: <err>` with a single `Caused by: <err>` instead of the duplicated path.

---
Written by an agent (Claude Code, claude-opus-4-7).


---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/npmrc/src/workspace_yaml.rs [hunk 4]
```diff
@@ -77,9 +97,24 @@ impl WorkspaceSettings {
...
+            && error.kind() == ErrorKind::NotFound
+        {
+            return Ok(None);
+        }
+
+        let settings: WorkspaceSettings = read_result
+            .map_err(|source| LoadWorkspaceYamlError::ReadFile { path: path.clone(), source })?
+            .pipe_as_ref(serde_saphyr::from_str)
+            .map_err(Box::new)
+            .map_err(|source| LoadWorkspaceYamlError::ParseYaml { path: path.clone(), source })?;
```
[the remark this answers is not in the document]
Comment (zkochan) [reply crates/npmrc/src/workspace_yaml.rs]: @KSXGitHub this one wasn't addressed.

[the remark this answers is not in the document]
Comment (KSXGitHub) [reply crates/npmrc/src/workspace_yaml.rs]: Fixed in 578f6de. `find_and_load` now iterates `start_dir.ancestors()` directly: at each level it reads `pnpm-workspace.yaml`, treats only `ErrorKind::NotFound` as "no manifest here, walk up", and propagates every other error (`EISDIR`, permission denied, etc.) through `LoadWorkspaceYamlError::ReadFile`. This also closes the TOCTOU window for free since the `is_file()` pre-check is gone. Added `find_propagates_when_manifest_path_is_a_directory` to pin the contract.

`find_workspace_manifest` stays as a private helper because `workspace_root_or` still uses it; only `find_and_load` switches to the new shape.

---
Written by an agent (Claude Code, claude-opus-4-7).


---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by zkochan — APPROVED** [review]
